webhook/workspace/handler/workspace.go

```diff
@@ -155,14 +165,22 @@ func (h *WebhookHandler) MutateWorkspaceV1alpha2OnUpdate(ctx context.Context, re
 			newWksp.Labels = map[string]string{}
 		}
 		newWksp.Labels[constants.DevWorkspaceCreatorLabel] = oldCreator
-		return h.returnPatched(req, newWksp)
+		response := h.returnPatched(req, newWksp)
+		if warnings != "" {
+			return response.WithWarnings(warnings)
+		}
+		return response
 	}
 
 	if newCreator != oldCreator {
 		return admission.Denied(fmt.Sprintf("label '%s' is assigned once devworkspace is created and is immutable", constants.DevWorkspaceCreatorLabel))
 	}
 
-	return admission.Allowed("new workspace has the same devworkspace as old one")
+	response := admission.Allowed("new workspace has the same devworkspace as old one")
+	if warnings != "" {
+		response.WithWarnings(warnings)
+	}
+	return response
```

Comment:
Not really important, but if the workspaces are identical here (not sure if that's _actually_ what we're checking...) then warnings should always be empty, right?

Reply:
The case we're checking here is "No change was made that was denied or required a patch, but features that are unsupported (and cause warnings) may have been added".

We need to modify all the cases where a response is returned because the warnings need to be added _in addition_ to the final response. If we were to immediately return a response with warnings (`admission.allowed().withWarnings(warnings)`), then we could allow other changes to be made that we should be denying (e.g. changing the devworkspace creator label)

Reply:
I just realized why this looks weird, my bad :facepalm:.
This should be changed to something like:
```go
	if warnings != "" {
		return admission.Allowed("").WithWarnings(warnings)
	}
	return admission.Allowed("new workspace has the same devworkspace as old one")
```

It doesn't make sense to say "new workspace has the same devworkspace as old one" when there are warnings created because of the new devworkspace being changed to have additional, unsupported features.

Though I'm not sure if responding with a blank message, i.e. `admission.Allowed("")` is correct.

Reply:
I updated my PR according to my comment.

It seems weird to have a blank reason/message, but checking the [implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/webhook/admission/response.go#L30) of `admission.Allowed()` shows that [if an empty reason is given, then it is omitted from the response](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/webhook/admission/response.go#L76).  So this seems valid.

Reply:
@amisevsk yes exactly - that's what I had realized :P thanks for catching that :+1: